src/Websites/Websites/Cmdlets/WebApps/PublishAzureWebApp.cs

```diff
@@ -72,6 +72,8 @@ public override void ExecuteCmdlet()
                 using (var s = File.OpenRead(ArchivePath))
                 {
                     HttpClient client = new HttpClient();
+                    // Considering the deployment of large packages the default time(150 seconds) is not sufficient. So increased the timeout by 300 seconds.
+                    client.Timeout = TimeSpan.FromMilliseconds(300000);
```

Comment:
Instead of changing the default timeout can we add a new optional property to the command to pass the timeout as an input , just like we already do for CLI Zip deployment command so that the user can change this to whatever value they want instead we hardcoding a value. Thanks!

Reply:
I agree. It's sometimes useful for large payload or pool network.
Let me know if I may proceed merging the PR or you'd like to implement timeout param here

Reply:
@panchagnula - made the code changes as suggested by adding an optional parameter.  Please review.